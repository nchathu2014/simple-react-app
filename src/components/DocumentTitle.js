import React,{Component} from 'react';

class DocumentTitle extends Component{

    constructor(props){
        super(props);
        this.state={
          count:0
        };
        this.handleOnClick = this.handleOnClick.bind(this);
    }

    handleOnClick(){
        this.setState({
            count:++(this.state.count)
        })
    }

    render(){

        let {count} = this.state;

        return(
            <div>
                <h1>{this.props.title}</h1>
                <button onClick={this.handleOnClick}>Click Me</button>
                <h4>{count}</h4>
            </div>
        );
    }

}

export default DocumentTitle;