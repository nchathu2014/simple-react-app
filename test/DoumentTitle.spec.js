import React from 'react';
import expect from 'expect';
import DocumentTitle from './../src/components/DocumentTitle';
import {shallow,mount} from 'enzyme';

const wrapper = shallow(
    <DocumentTitle title="hello" name="nuwan"/>
);

test('render DoumentTitle',()=>{
    expect(wrapper).toMatchSnapshot();
});

test('render a document title',()=>{
    expect(wrapper.instance().props.title).toEqual('hello');
    expect(wrapper.instance().props.name).toEqual('nuwan');
});

test('Simulate onClick',()=>{
    const count=0;
    const mockedEvent = { target: {} }
    const wrapper = mount(
        <DocumentTitle title="hello" name="nuwan"/>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find('button').simulate('click',mockedEvent);
});