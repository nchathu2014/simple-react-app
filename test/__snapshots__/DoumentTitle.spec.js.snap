// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulate onClick 1`] = `
<DocumentTitle
  name="nuwan"
  title="hello"
>
  <div>
    <h1>
      hello
    </h1>
    <button
      onClick={[Function]}
    >
      Click Me
    </button>
    <h4>
      0
    </h4>
  </div>
</DocumentTitle>
`;

exports[`render DoumentTitle 1`] = `
<div>
  <h1>
    hello
  </h1>
  <button
    onClick={[Function]}
  >
    Click Me
  </button>
  <h4>
    0
  </h4>
</div>
`;
